facebook-commerce.php

```diff
@@ -785,26 +790,26 @@ private function find_variation_product_item_ids( $product_group_id ) {
 		return $product_item_ids;
 	}
 
-
 	/**
-	 * Gets the total of published products.
+	 * Gets the total number of published products.
 	 *
 	 * @return int
 	 */
 	public function get_product_count() {
+		$product_counts = wp_count_posts( 'product' );
+		return $product_counts->publish;
+	}
 
-		$args = array(
-			'post_type'      => 'product',
-			'post_status'    => 'publish',
-			'posts_per_page' => -1,
-			'fields'         => 'ids',
-		);
-
-		$products = new WP_Query( $args );
-
-		wp_reset_postdata();
 
-		return $products->found_posts;
+	/**
+	 * Should full batch-API sync be allowed?
+	 *
+	 * Used to disable various full sync UI/APIs to avoid performance impact.
+	 *
+	 * @return boolean True if full sync is safe.
+	 */
+	public function allow_full_batch_api_sync() {
+		return ( ( $this->get_product_count() < self::MAX_PRODUCTS_FOR_FULL_SYNC ) && ! \WC_Facebookcommerce_Utils::is_Atomic_env() );
```

Comment:
Ideally we'd expose a mechanism that any hosting platform (or site) could use, rather than hard-code this specifically for atomic platform. We can probably do that by exposing an option/filter, and working with Atomic platform team to deploy to all sites.